crates/y-sweet-worker/src/config.rs

```diff
@@ -27,12 +85,24 @@ impl From<&Env> for Configuration {
                 })
                 .unwrap_or(45),
         );
-        Self {
+
+        let bucket_kind = env.var(BUCKET_KIND).map_or_else(
+            |_| Ok(BucketKind::NativeR2),
+            |b| BucketKind::from_str(&b.to_string()),
+        )?;
+        let s3_config = if let BucketKind::S3Compatible = bucket_kind {
+            Some(parse_s3_config(&env).unwrap())
```

Comment:
I'm generally ok with panicking if it's a config error, but since `parse_s3_config` returns an anyhow error already, let's just use a `?` here.